homeassistant/components/zha/config_flow.py

```diff
@@ -53,48 +224,306 @@ async def async_step_user(self, user_input=None):
+            data_schema=vol.Schema(
+                {
+                    vol.Required(UPLOADED_BACKUP_FILE): FileSelector(
+                        FileSelectorConfig(accept=".json,application/json")
+                    )
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_choose_automatic_backup(self, user_input=None):
+        """Choose an automatic backup."""
+        if self.show_advanced_options:
+            # Always show the PAN IDs when in advanced mode
+            choices = [
+                _format_backup_choice(backup, pan_ids=True) for backup in self._backups
+            ]
+        else:
+            # Only show the PAN IDs for multiple backups taken on the same day
+            num_backups_on_date = collections.Counter(
+                backup.backup_time.date() for backup in self._backups
+            )
+            choices = [
+                _format_backup_choice(
+                    backup, pan_ids=(num_backups_on_date[backup.backup_time.date()] > 1)
+                )
+                for backup in self._backups
+            ]
+
+        if user_input is not None:
+            index = choices.index(user_input[CHOOSE_AUTOMATIC_BACKUP])
+            self._chosen_backup = self._backups[index]
+
+            return await self.async_step_maybe_confirm_ezsp_restore()
+
+        return self.async_show_form(
+            step_id="choose_automatic_backup",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CHOOSE_AUTOMATIC_BACKUP, default=choices[0]): vol.In(
+                        choices
+                    ),
+                }
+            ),
+        )
+
+    async def async_step_maybe_confirm_ezsp_restore(self, user_input=None):
+        """Confirm restore for EZSP radios that require permanent IEEE writes."""
+        if self._radio_type != RadioType.ezsp:
+            await self._restore_backup(self._chosen_backup)
+            return await self._async_create_radio_entity()
+
+        # We have no way to partially load network settings if no network is formed
+        if self._current_settings is None:
+            # Since we are going to be restoring the backup anyways, write it to the
+            # radio without overwriting the IEEE but don't take a backup with these
+            # temporary settings
+            temp_backup = _prevent_overwrite_ezsp_ieee(self._chosen_backup)
+            await self._restore_backup(temp_backup, create_new=False)
+            await self._async_load_network_settings()
+
+        if (
+            self._current_settings.node_info.ieee == self._chosen_backup.node_info.ieee
+            or not self._current_settings.network_info.metadata["ezsp"][
+                "can_write_custom_eui64"
+            ]
+        ):
+            # No point in prompting the user if the backup doesn't have a new IEEE
+            # address or if there is no way to overwrite the IEEE address a second time
```

Comment:
If there is no way to overwrite the IEEE address, what is the effect of restoring a backup? If it won't work, we should abort maybe or show an error and pick a new option?